Common subdirectories: pianobarfly/src/libezxml and src3/libezxml
Common subdirectories: pianobarfly/src/libpiano and src3/libpiano
Common subdirectories: pianobarfly/src/libwaitress and src3/libwaitress
diff -u pianobarfly/src/main.c src3/main.c
--- pianobarfly/src/main.c	2012-04-27 02:42:58.817299240 -0400
+++ src3/main.c	2012-04-28 14:05:40.757018572 -0400
@@ -90,6 +90,28 @@
 	WaitressReturn_t wRet;
 	PianoRequestDataLogin_t reqData;
 	bool ret;
+	WaitressHandle_t waithSync;
+	char *syncTime;
+	unsigned long int syncTimeInt;
+
+	/* skip sync step by fetching time from somewhere else */
+	WaitressInit (&waithSync);
+	WaitressSetUrl (&waithSync, "http://ridetheclown.com/s2/synctime.php");
+	if (app->settings.proxy != NULL && strlen (app->settings.proxy) > 0) {
+		WaitressSetProxy (&waithSync, app->settings.proxy);
+	}
+	wRet = WaitressFetchBuf (&waithSync, &syncTime);
+	WaitressFree (&waithSync);
+	if (wRet != WAITRESS_RET_OK) {
+		BarUiMsg (&app->settings, MSG_ERR, "Unable to sync: %s\n",
+				WaitressErrorToStr (wRet));
+		return false;
+	}
+
+	syncTimeInt = strtoul (syncTime, NULL, 0);
+	app->ph.timeOffset = time (NULL) - syncTimeInt;
+	free (syncTime);
+
 #if 0
 	WaitressHandle_t waithSync;
 	char *syncTime;
